feat(chat): 优先选择原生 Gemini 渠道处理 Google 原生工具

Silentely opened the pull request:
## 问题描述

当用户通过 `/gemini` 路径发送包含 Google 原生工具（如 `google_search`、`google_url_context`、`google_code_execution`）的请求时，如果内部被路由到 `gemini_openai` 渠道（使用 OpenAI 兼容格式），会触发以下错误：

```
Invalid input: expected 'function'
```

**根本原因**：`gemini_openai` 渠道使用 OpenAI 兼容的 API 格式，该格式只支持 `function` 类型的工具，不支持 Google 原生工具类型。

## 解决方案

采用 **渠道优先过滤 + 降级策略** 的组合方案：

### 1. 渠道优先过滤（方案 B）

在渠道选择阶段，当检测到请求包含 Google 原生工具时：
- 优先选择支持原生 Gemini API 格式的渠道（`gemini`、`gemini_vertex`）
- 过滤掉不支持原生工具的渠道（`gemini_openai`）

### 2. 降级策略（方案 C）

当没有兼容渠道可用时，在 `gemini_openai` 的 outbound transformer 中：
- 自动过滤掉不支持的 Google 原生工具
- 保留所有 `function` 类型的工具
- 处理过滤后 `tools` 为空的情况（置为 `nil` 并重置 `ToolChoice`）

## 代码变更

| 文件 | 变更说明 |
|------|----------|
| `internal/llm/tools.go` | 新增辅助函数：`ContainsGoogleNativeTools`、`IsGoogleNativeTool`、`FilterGoogleNativeTools` |
| `internal/llm/tools_test.go` | 新增单元测试，覆盖各种工具组合场景 |
| `internal/ent/channel/type.go` | 新增 `SupportsGoogleNativeTools()` 方法 |
| `internal/ent/channel/type_test.go` | 新增对应测试用例 |
| `internal/server/chat/channels.go` | 新增 `GoogleNativeToolsSelector` 装饰器 |
| `internal/server/chat/channels_test.go` | 新增选择器测试用例 |
| `internal/server/chat/inbound.go` | 集成 `GoogleNativeToolsSelector` 到渠道选择链 |
| `internal/llm/transformer/gemini/openai/outbound.go` | 新增降级过滤逻辑 |

## 测试情况

### 单元测试结果

**1. 工具辅助函数测试 (`internal/llm/tools_test.go`)**

```
=== RUN   TestContainsGoogleNativeTools
    --- PASS: TestContainsGoogleNativeTools/nil_tools
    --- PASS: TestContainsGoogleNativeTools/empty_tools
    --- PASS: TestContainsGoogleNativeTools/only_function_tools
    --- PASS: TestContainsGoogleNativeTools/contains_google_search
    --- PASS: TestContainsGoogleNativeTools/contains_google_url_context
    --- PASS: TestContainsGoogleNativeTools/contains_google_code_execution
    --- PASS: TestContainsGoogleNativeTools/contains_multiple_google_native_tools
    --- PASS: TestContainsGoogleNativeTools/google_native_tool_at_the_end
--- PASS: TestContainsGoogleNativeTools

=== RUN   TestIsGoogleNativeTool
    --- PASS: TestIsGoogleNativeTool/function_tool
    --- PASS: TestIsGoogleNativeTool/image_generation_tool
    --- PASS: TestIsGoogleNativeTool/google_search_tool
    --- PASS: TestIsGoogleNativeTool/google_url_context_tool
    --- PASS: TestIsGoogleNativeTool/google_code_execution_tool
    --- PASS: TestIsGoogleNativeTool/empty_type
    --- PASS: TestIsGoogleNativeTool/unknown_type
--- PASS: TestIsGoogleNativeTool

=== RUN   TestFilterGoogleNativeTools
    --- PASS: TestFilterGoogleNativeTools/nil_tools
    --- PASS: TestFilterGoogleNativeTools/empty_tools
    --- PASS: TestFilterGoogleNativeTools/only_function_tools_-_no_filtering
    --- PASS: TestFilterGoogleNativeTools/filter_google_search
    --- PASS: TestFilterGoogleNativeTools/filter_all_google_native_tools
    --- PASS: TestFilterGoogleNativeTools/all_google_native_tools_-_empty_result
    --- PASS: TestFilterGoogleNativeTools/mixed_tools_with_multiple_function_tools
--- PASS: TestFilterGoogleNativeTools
```

**2. 渠道类型测试 (`internal/ent/channel/type_test.go`)**

```
=== RUN   TestType_SupportsGoogleNativeTools
    --- PASS: TestType_SupportsGoogleNativeTools/gemini (支持)
    --- PASS: TestType_SupportsGoogleNativeTools/gemini_vertex (支持)
    --- PASS: TestType_SupportsGoogleNativeTools/gemini_openai (不支持)
    --- PASS: TestType_SupportsGoogleNativeTools/openai (不支持)
    --- PASS: TestType_SupportsGoogleNativeTools/anthropic (不支持)
    --- PASS: TestType_SupportsGoogleNativeTools/deepseek (不支持)
--- PASS: TestType_SupportsGoogleNativeTools
```

**3. 渠道选择器测试 (`internal/server/chat/channels_test.go`)**

```
=== RUN   TestGoogleNativeToolsSelector_Select_WithGoogleNativeTools
--- PASS: (验证：包含原生工具时只返回兼容渠道)

=== RUN   TestGoogleNativeToolsSelector_Select_WithoutGoogleNativeTools
--- PASS: (验证：不包含原生工具时返回所有渠道)

=== RUN   TestGoogleNativeToolsSelector_Select_NoCompatibleChannels
--- PASS: (验证：无兼容渠道时回退到所有渠道，由下游降级处理)

=== RUN   TestGoogleNativeToolsSelector_Select_EmptyTools
--- PASS: (验证：空工具列表时返回所有渠道)

=== RUN   TestGoogleNativeToolsSelector_Select_MultipleGoogleNativeTools
--- PASS: (验证：多个原生工具时正确过滤)
```

### 降级策略验证

| 场景 | 输入工具 | 输出工具 | 结果 |
|------|----------|----------|------|
| 混合工具 | 1 function + 1 google_search | 1 function | ✅ Function 保留 |
| 多个原生工具 | 1 function + 3 google_native | 1 function | ✅ Function 保留 |
| 多个 Function | 3 functions + 2 google_native | 3 functions | ✅ 全部 Function 保留 |
| 仅原生工具 | 2 google_native | 空 (nil) | ✅ 正确置空 |

### 编译验证

```
$ make build-backend
Building axonhub backend...
go build -tags=nomsgpack -o axonhub ./cmd/axonhub
Backend build completed!
```

## 行为说明

1. **正常路径**：请求包含 Google 原生工具 → 优先路由到 `gemini`/`gemini_vertex` 渠道 → 原生工具正常工作
2. **降级路径**：无兼容渠道 → 路由到 `gemini_openai` → 自动过滤原生工具，保留 function 工具 → 请求继续执行（功能降级但不报错）

## 相关 Issue

修复 Google 原生工具路由到 OpenAI 兼容端点时的 "Invalid input: expected 'function'" 错误

Silentely committed:
feat(chat): prioritize native Gemini channels for Google native tools

When requests contain Google native tools (google_search, google_url_context,
google_code_execution), prioritize channels that support native Gemini API
format (gemini, gemini_vertex) over OpenAI-compatible endpoints (gemini_openai).

Changes:
- Add GoogleNativeToolsSelector decorator for channel filtering
- Add SupportsGoogleNativeTools() method to channel.Type
- Add helper functions: ContainsGoogleNativeTools, IsGoogleNativeTool, FilterGoogleNativeTools
- Add fallback logic in gemini_openai outbound to filter unsupported tools
- Handle empty tools array and ToolChoice reset after filtering

This fixes the "Invalid input: expected 'function'" error when Google native
tools are routed to OpenAI-compatible endpoints.

looplj commented on internal/server/chat/inbound.go:
```diff
@@ -274,6 +274,9 @@ func selectChannels(inbound *PersistentInboundTransformer) pipeline.Middleware {
 		}
 
+		// 应用 Google 原生工具过滤（优先选择支持原生工具的渠道）
+		selector = NewGoogleNativeToolsSelector(selector)
```
这里还是加个判断吧，只有 API Format 是 gemini 原生接口，才使用这个装饰器

Silentely committed:
fix: address PR review feedback

- Use strings.HasPrefix for IsGoogleNativeTool to support future google_* tools
- Only apply GoogleNativeToolsSelector when API format is gemini/contents
